go-controller/pkg/ovn/pods.go

```diff
@@ -283,103 +254,126 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
+
+		// If the pod already has annotations use the existing static
+		// IP/MAC from the annotation.
+		args = append(args,
+			"--", "lsp-set-addresses", portName, fmt.Sprintf("%s %s", podMac, annotation.IP.IP),
+			"--", "--if-exists", "clear", "logical_switch_port", portName, "dynamic_addresses",
+		)
+	} else {
+		gatewayCIDR, _ = util.GetNodeWellKnownAddresses(nodeSubnet)
+
+		addresses := "dynamic"
+		networks, err := util.GetPodNetSelAnnotation(pod, util.DefNetworkAnnotation)
+		if err != nil || (networks != nil && len(networks) != 1) {
+			return fmt.Errorf("error while getting custom MAC config for port %q from "+
+				"default-network's network-attachment: %v", portName, err)
+		} else if networks != nil && networks[0].MacRequest != "" {
+			klog.V(5).Infof("Pod %s/%s requested custom MAC: %s", pod.Namespace, pod.Name, networks[0].MacRequest)
+			addresses = networks[0].MacRequest + " dynamic"
 		}
-		oc.logicalPortCache[portName] = logicalSwitch
-		return oc.addPodToNamespace(pod.Namespace, annotation.IP.IP, portName)
-	}
 
-	addressStr := "dynamic"
-	networks, err := util.GetPodNetSelAnnotation(pod, util.DefNetworkAnnotation)
-	if err != nil || (networks != nil && len(networks) != 1) {
-		return fmt.Errorf("error while getting custom MAC config for port %q from "+
-			"default-network's network-attachment: %v", portName, err)
-	} else if networks != nil && networks[0].MacRequest != "" {
-		klog.V(5).Infof("Pod %s/%s requested custom MAC: %s", pod.Namespace, pod.Name, networks[0].MacRequest)
-		addressStr = networks[0].MacRequest + " dynamic"
+		// If it has no annotations, let OVN assign it IP and MAC addresses
+		args = []string{
+			"--may-exist", "lsp-add", logicalSwitch, portName,
+			"--", "lsp-set-addresses", portName, addresses,
+		}
 	}
-	out, stderr, err = util.RunOVNNbctl(
-		"--", "--may-exist", "lsp-add", logicalSwitch, portName,
-		"--", "lsp-set-addresses", portName, addressStr,
-		"--", "set", "logical_switch_port", portName, "external-ids:namespace="+pod.Namespace,
-		"external-ids:logical_switch="+logicalSwitch, "external-ids:pod=true")
+	args = append(args, "--", "set", "logical_switch_port", portName, "external-ids:namespace="+pod.Namespace, "external-ids:pod=true")
+
+	out, stderr, err = util.RunOVNNbctl(args...)
 	if err != nil {
 		return fmt.Errorf("Error while creating logical port %s stdout: %q, stderr: %q (%v)",
 			portName, out, stderr, err)
 	}
 
-	oc.logicalPortCache[portName] = logicalSwitch
-
-	gatewayIPnet, _ := util.GetNodeWellKnownAddresses(nodeSubnet)
+	// If the pod has not already been assigned addresses, read them now
+	if podMac == nil || podCIDR == nil {
+		var podIP net.IP
+		podMac, podIP, err = waitForPodAddresses(portName)
+		if err != nil {
+			return err
+		}
+		podCIDR = &net.IPNet{IP: podIP, Mask: nodeSubnet.Mask}
+	}
 
-	podMac, podIP, err := waitForPodAddresses(portName)
+	var uuid string
+	uuid, _, err = util.RunOVNNbctl("get", "logical_switch_port", portName, "_uuid")
```

Comment:
It looks like you are trying to get  the uuid here. Is there no way to obtain it from the actual cmd that creates the port on line 305?

Reply:
@trozet https://bugzilla.redhat.com/show_bug.cgi?id=1806788

There might be some fun stuff we can do (and Girish had a suggestion) but that bug has the details of why I broke this out.

Reply:
OK makes sense, but can you add a comment here with that bz so we know to remove this when that gets fixed? Otherwise patch lgtm

Reply:
@trozet done